### PR TITLE
Implement checkbox field

### PR DESCRIPTION
**Pull request by IgorZiegler**
Create Checkbox block which use jquery.styler plugin. Add example with checkbox block usage. Add precommit script with eslint. Remove resolve from webpack.confing because it's identically with default settings. Remove DefinePlugin.

fixes #20

### CODE REVIEW

File: frontend/blocks/checkbox/checkbox.scss [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+.checkbox label,
+.radio label {
+  padding-left: 30px;
+  position: relative;
+}
+.checkbox .jq-checkbox,
+.radio .jq-radio {
+  position: absolute !important;
+  top: 0;
+}
```
Comment (amenshov) [inline frontend/blocks/checkbox/checkbox.scss]: no new line at the end of file

File: frontend/blocks/checkbox/checkbox.scss [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  transition: all 0.1s ease-out;
+}
+.jq-checkbox.focused,
+.jq-radio.focused {
+  border-color: #0a9acb;
+}
+.jq-checkbox.disabled,
+.jq-radio.disabled {
+  opacity: 0.55;
+  filter: alpha(opacity=55.00000000000001);
```
Comment (amenshov) [inline frontend/blocks/checkbox/checkbox.scss]: why do we need such a high precision?

File: frontend/blocks/checkbox/checkbox.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import template from './checkbox.jade';
+
+import Base from '../base/base';
+import Factory from '../factory';
+
+export default class Checkbox extends Base {
+  get templateFn() {
+    return template;
+  }
+  render() {
```
Comment (amenshov) [inline frontend/blocks/checkbox/checkbox.js]: add empty line between methods

File: frontend/blocks/checkbox/checkbox.jade [hunk 4]
```diff
@@ -0,0 +1,10 @@
+extends ../base/base.jade
+
+block inner
+    div(class="#{checkboxCls}-checkbox #{cls}")
+        .checkbox
+            div(class="#{checkboxCls}-checkbox-wrapper")
+                label
+                    input(type='checkbox', checked=(checked===true ? "checked" : undefined) class=checkboxCls)
```
Comment (amenshov) [inline frontend/blocks/checkbox/checkbox.jade]: Are you sure that you cannot simply write `checked=checked`?

File: frontend/blocks/checkbox/checkbox.scss [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  border-radius: 2px;
+  &.checked .jq-checkbox__div {
+    width: 14px;
+  }
+}
+
+.checkbox .jq-checkbox {
+  position: absolute !important;
+  top: 0;
+}
```
Comment (102) [inline frontend/blocks/checkbox/checkbox.scss]: No newline at the end of file

**Pull request merged**